code/platforms/win/app.py

```diff
@@ -1,11 +1,47 @@
 # defines the default app actions for windows
 
-from talon import Context, actions
+import win32gui
+from talon import Context, actions, ui
 ctx = Context()
 ctx.matches = r"""
 os: windows
 """
 
+# adapted this from code posted by @PeterLinder
+# return list of top level windows
+def get_top_level_windows():
+    def enumHandler(hwnd, resultList):
+        resultList.append(hwnd)
+
+    top_level_windows = []
+
+    win32gui.EnumWindows(enumHandler, top_level_windows)
+    return top_level_windows
+
+# focuses the n-th next or n-th previous instance of the current
+# app, depending on the magnitude and sign of the 'direction' arg
```

Comment:
this should be a docstring instead of a comment but w/e

Reply:
Done.